File: frontend/src/types/graph.ts
export interface NodeData {
  id: string;
  label: string;
  properties: Record<string, unknown>;
}

export interface RelationshipData {
  id: string;
  type: string;
  source_id: string;
  target_id: string;
  properties: Record<string, unknown>;
}

export interface GraphData {
  nodes: NodeData[];
  relationships: RelationshipData[];
}

File: backend/app/schemas/graph.py
from typing import Any

from pydantic import BaseModel


class NodeCreate(BaseModel):
    label: str
    properties: dict[str, Any] = {}


class NodeUpdate(BaseModel):
    label: str | None = None
    properties: dict[str, Any] = {}


class RelationshipCreate(BaseModel):
    source_id: str
    target_id: str
    type: str
    properties: dict[str, Any] = {}


class RelationshipUpdate(BaseModel):
    type: str | None = None
    properties: dict[str, Any] = {}


class NodeResponse(BaseModel):
    id: str
    label: str
    properties: dict[str, Any]


class RelationshipResponse(BaseModel):
    id: str
    type: str
    source_id: str
    target_id: str
    properties: dict[str, Any]


class GraphData(BaseModel):
    nodes: list[NodeResponse]
    relationships: list[RelationshipResponse]
